File: backend/src/models/Person.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IPerson extends Document {
  userId: mongoose.Types.ObjectId;
  name: string;
  email?: string;
  phone?: string;
  company?: string;
  notes?: string;
  tags: string[];
  linkedNotes: mongoose.Types.ObjectId[];
  meetings: mongoose.Types.ObjectId[];
  reminders: mongoose.Types.ObjectId[];
  createdAt: Date;
  updatedAt: Date;
}

const personSchema = new Schema<IPerson>(
  {
    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    name: { type: String, required: true },
    email: { type: String },
    phone: { type: String },
    company: { type: String },
    notes: { type: String },
    tags: [{ type: String }],
    linkedNotes: [{ type: Schema.Types.ObjectId, ref: 'Note' }],
    meetings: [{ type: Schema.Types.ObjectId, ref: 'Meeting' }],
    reminders: [{ type: Schema.Types.ObjectId, ref: 'Reminder' }],
  },
  { timestamps: true }
);

personSchema.index({ userId: 1, name: 1 });
personSchema.index({ email: 1 });

export const Person = mongoose.model<IPerson>('Person', personSchema);

File: backend/models/person.py
"""
Person/Contact model for relationship management
"""

from beanie import Document, PydanticObjectId
from pydantic import Field, EmailStr
from typing import List, Optional
from datetime import datetime


class Person(Document):
    """Person/Contact document model"""
    
    user_id: PydanticObjectId
    name: str
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    company: Optional[str] = None
    role: Optional[str] = None
    tags: List[str] = Field(default_factory=list)
    notes: Optional[str] = None
    linked_notes: List[PydanticObjectId] = Field(default_factory=list)
    linked_meetings: List[PydanticObjectId] = Field(default_factory=list)
    last_contact: Optional[datetime] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    class Settings:
        name = "people"
        indexes = [
            "user_id",
            "name",
            [("user_id", 1), ("name", 1)],
        ]
    
    class Config:
        json_schema_extra = {
            "example": {
                "name": "Sarah Johnson",
                "email": "sarah@example.com",
                "company": "Tech Corp",
                "role": "Product Manager",
                "tags": ["colleague", "project-lead"]
            }
        }
